### ISSUE TITLE
Crashes on Windows loading gdsqlite via GDNative (works in 3.0, crash in 3.0.6)

### ISSUE THREAD

**Issue by torohaifisch** [godotengine/godot#21346]
**Godot version:**
godot 3.0.6

**OS/device including version:**
windows 10 / gtx1060

**Issue description:**
godot closes itself after a while

**Steps to reproduce:**
i am not really sure just happens randomly after a few seconds or minutes even if i open and leave it alone

was using godot 3.0 without any issue and decided to try 3.0.6 but this started to happen 
idk if there is a way to get some crash dumps?

console shows this error:

```
ERROR: read: Server disconnected!

   At: drivers/windows/stream_peer_tcp_winsock.cpp:203
```

**Comment by malbach** [comment]
Hi,
Since it's Windows 10, may I assume you're talking about Godot 3.0.6 win64 stable ?
I'm using it too and no problem here, on Windows 10 and GTX 1050 Ti (Mobile version, I'm on a laptop)
When you say

> even if i open and leave it alone

do you mean you don't even open a project ?

**Comment by VitaZheltyakov** [comment]
`Server disconnected!` - this is a [popular error](https://github.com/godotengine/godot/issues?q=is%3Aissue+Server+disconnected+is%3Aopen) caused by a bad caching system. Developers usually ignore this problem.

**Comment by akien-mga** [comment]
@VitaZheltyakov Again, this is not true, as I explained in https://github.com/godotengine/godot/issues/21268#issuecomment-415460653. This error is just a consequence of the crash, it tells us nothing about what caused the crash itself.

**Comment by akien-mga** [comment]
What plugin is it? Can you provide an example project that triggers the bug? If it worked in 3.0 and not in 3.0.6, that's a bug we need to fix.

**Comment by VitaZheltyakov** [comment]
@akien-mga See:
https://github.com/godotengine/godot/issues/15369

If you connect to the host, then manually disconnect and try to send a request, you will get an error. The application will crash. This error causes the application to crash.

The error occurs because the connection status is checked only once. Then it is cached.

**Comment by torohaifisch** [comment]
@akien-mga  plugin is [gdsqlite-native](https://github.com/khairul169/gdsqlite-native) release version 1.2
at least for me it starts crashing when i preload the file
 `const SQLite = preload("res://lib/gdsqlite/gdsqlite.gdns");
`

**Comment by Zireael07** [comment]
GDNative has changed between 3.0 and 3.0.6, so it's not surprising that an older version of the plugin doesn't work.

**Comment by akien-mga** [comment]
@karroffel WDYT about this issue? Is there a compat breakage that we did/should fix in 3.0.x, or is it a downstream issue?

**Comment by karroffel** [comment]
@akien-mga It seems like the plugin is saving the `GDNativeLibrary` as as `.tres` file, which was unsupported in 3.0 but was patched later on, so it's already fixed in that it's working on the latest stable.
